electron/js/lib/openGraph.js

```diff
@@ -41,21 +42,38 @@ const fetchImageAsBase64 = url => {
 };
 
 const fetchOpenGraphData = url => {
-  const isHtmlContentTypePromise = new Promise(resolve => {
-    const headerRequest = request.head(url);
-    headerRequest.on('response', response => {
-      const isHtmlContentType =
-        response.headers['content-type'] && response.headers['content-type'].match(/.*text\/html/);
-      resolve(isHtmlContentType);
+  const CONTENT_TYPE_LIMIT = 1000000; // 1MB
+  const parsedUrl = urlUtil.parse(url);
+  const normalizedUrl = parsedUrl.protocol ? parsedUrl : urlUtil.parse(`http://${url}`);
+
+  return new Promise((resolve, reject) => {
+    const getContentRequest = request.get(urlUtil.format(normalizedUrl), (error, response, body) => {
+      return error ? reject(error) : resolve(body);
     });
-  });
 
-  return isHtmlContentTypePromise.then(isHtmlContentType => {
-    if (isHtmlContentType) {
-      return new Promise((resolve, reject) => openGraph(url, (error, meta) => (error ? reject(error) : resolve(meta))));
-    }
-    throw new Error('Unhandled format for open graph generation');
-  });
+    getContentRequest.on('response', ({headers}) => {
+      const contentType = headers['content-type'] || '';
+      const isHtmlContentType = contentType.match(/.*text\/html/);
+
+      if (!isHtmlContentType) {
+        getContentRequest.abort();
+        throw new Error(`Unhandled format for open graph generation ('${contentType}')`);
+      }
+    });
+
+    let requestCurrentSize = 0;
+    let partialBody = '';
+    getContentRequest.on('data', buffer => {
+      const chunk = buffer.toString();
+      requestCurrentSize += buffer.length;
+      partialBody += chunk;
```

Comment:
Humm I would argue that this is not the same thing. `buffer.length` is the number of bytes while `body.length` is the number of characters (that could differ because of unicode).

Reply:
Exact but it does not need to be that precise ;)

Reply:
(If you feel you need to keep this precision then I would just suggest to rename `requestCurrentSize` to `responseCurrentSize`)

Reply:
The number of characters is good enough indeed